SPECS/kernel-mshv/kernel-mshv.spec

```diff
@@ -120,7 +121,15 @@ make INSTALL_MOD_PATH=%{buildroot} modules_install
 # This configuration contains additional boot parameters required in our
 # Linux-Dom0-based images. 
 mkdir -p %{buildroot}%{_sysconfdir}/default/grub.d
-install -m 750 %{SOURCE3} %{buildroot}%{_sysconfdir}/default/grub.d/50_mariner_mshv.cfg
+mkdir -p %{buildroot}%{_sysconfdir}/grub.d
+
+# Initially install this configuration script WITHOUT execution bits.
```

Comment:
@ddstreetmicrosoft I understand your comment about using the execution bits to control which bash scripts are sourced is fundamentally wrong practice.

Given that, what is a good way to enforce this behavior: install the /etc/default/grub.d configuration for dom0 but leave it in a dormant state.

We could simply not worry about this, assuming that any system which installs kernel-mshv has the proprietary components (lxhvloader, hvix, etc). The downside to that is that anyone who accidentally installs kernel-mshv will be implicitly using the kernel-mshv kernel.

Reply:
Note that the dom0 menuentry still boots successfully without the proprietary components. Grub will be confused for an extra few seconds during startup however and we will boot into the kernel-mshv version of the kernel.

Reply:
Could the script check if it is "valid" on its own? What information do we have to decide if we should  use dom0 or not?